e2e/wdio/core/tests/dialog.e2e-spec.ts

```diff
@@ -0,0 +1,676 @@
+
+            for (let i = 0; i < dialogCount; i++) {
+                scrollIntoView(dialogExamples + button, i);
+                addIsActiveClass(dialogExamples + button, i);
+                saveElementScreenshot(dialogExamples + button, `dialog-${i}-example-active-state-${getImageTagBrowserPlatform()}-`, dialogPage.getScreenshotFolder(), i);
+                expect(checkElementScreenshot(dialogExamples + button, `dialog-${i}-example-active-state-${getImageTagBrowserPlatform()}-`, dialogPage.getScreenshotFolder(), i))
+                    .toBeLessThan(5, `dialog ${i} active state screenshot doesn't match baseline`);
+            }
+        }, 1);
+
+        it('should check dialog button hover states', () => {
+            checkButtonState('hover', mouseHoverElement);
+        }, 1);
+
+        it('should check dialog button focus states', () => {
+            checkButtonState('focus', focusElement);
+        }, 1);
+
+        it('should check dialog button active states', () => {
+            checkButtonState('active', addIsActiveClass);
+        }, 1);
+
+        it('should check custom backdrop example', () => {
+            openDialog(customDialog);
+
+            saveElementScreenshot(dialog, `dialog-with-custom-backdrop-${getImageTagBrowserPlatform()}-`, dialogPage.getScreenshotFolder());
+            expect(checkElementScreenshot(dialog, `dialog-dialog-with-custom-backdrop-${getImageTagBrowserPlatform()}-`, dialogPage.getScreenshotFolder()))
+                .toBeLessThan(10, `dialog with custom backdrop screenshot doesn't match baseline`);
+
+            closeDialog();
+        });
+
+        function checkButtonState(stateCheck: string, applyStateFunction: any): void {
+            const dialogCount = getElementArrayLength(dialogExamples + button);
+
+            for (let i = 0; i < dialogCount; i++) {
+                if (i === 3 || i === 4 || i === 5) {
+                    // skip: sometimes dialog auto closes before screenshot taken
+                    continue;
+                }
+
+                click(dialogExamples + button, i);
+                waitForElDisplayed(dialog);
+                const dialogButtonCount = getElementArrayLength(dialog + button);
+
+                for (let j = 0; j < dialogButtonCount; j++) {
+                    applyStateFunction(dialog + button, j);
+                    saveElementScreenshot(dialog + button, `dialog-${i}-button-${j}-${stateCheck}-state-${getImageTagBrowserPlatform()}-`, dialogPage.getScreenshotFolder(), j);
+                    expect(checkElementScreenshot(dialog + button, `dialog-${i}-button-${j}-${stateCheck}-state-${getImageTagBrowserPlatform()}-`, dialogPage.getScreenshotFolder(), j))
+                        .toBeLessThan(5, `dialog ${i} button ${j} ${stateCheck} state screenshot doesn't match baseline`);
+                }
+                if (i === complexExample) {
+                    click(dialog + button, 2);
```

Comment:
Maybe better to create dialogCloseButton instead of `dialog + button`. 
dialogResizeHandle. 
I think here it is harder to read the concat all the time.

Reply:
well, the dialogs use different buttons for different actions. for example, for some (dialog + button, 0) is an 'x' btn, for others it's a clear btn, for another it's accept btn. so it doesn't seem logical to me to create the different selector names for the same selector. because I would have 
clearBtn = this.dialog + this.button 
closeBtn = this.dialog + this.button
interestedBtn = this.dialog + this.button
and then close btn would sometimes read as click(closeBtn, 2) which seems like there are 3 close buttons, which isn't true)